Update to support .NET Core 3.1

darknessxk opened the pull request:
Changed the BeginInvoke and EndInvoke calls to proper async Tasks usages

darknessxk committed:
Ported to NET Core

darknessxk committed:
Updated Readme

darknessxk committed:
Added proper async Tasks instead of using unsupported .NET Core BeginInvoke/EndInvoke

darknessxk committed:
Merge remote-tracking branch 'upstream/master' into master

SergioLuis commented:
It would be neat if this PR gets merged. Thank you @darknessxk for your help with this, as I've hit the same issue with this library.

dave7280 commented:
When can we expect this to be merged?

SergioLuis commented:
I don't think this will get merged, like ever :(

xas commented on websocket-sharp/websocket-sharp.csproj:
```diff
@@ -1,149 +1,8 @@
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
-    <SignAssembly>true</SignAssembly>
-    <AssemblyOriginatorKeyFile>websocket-sharp.snk</AssemblyOriginatorKeyFile>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Isn't more interesting to replace it with
```suggestion
    <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFramework>
```

apdevelop replied:
> Isn't more interesting to replace it with

Or maybe even better:
`<TargetFrameworks>netstandard2.0;net452</TargetFrameworks>`

jusoGames commented:
I have more than two years using websocketsharp in a .net core 3.1 project, the project is in production since then on Ubuntu 18.04, I have not suffered from any problems.

When I read about the project being incompatible it makes me very strange.

darknessxk committed:
Conflicts resolving for merging remote sta branch to katlab branch

# Conflicts:
#	Example/Example.csproj
#	Example/Program.cs
#	Example2/Chat.cs
#	Example2/Echo.cs
#	Example2/Program.cs
#	Example3/Chat.cs
#	Example3/Echo.cs
#	Example3/Program.cs
#	websocket-sharp.sln
#	websocket-sharp/Net/HttpStreamAsyncResult.cs
#	websocket-sharp/WebSocket.cs

darknessxk committed:
Merge pull request #2 from katsuhirogg/pulls/1335341197/1

Pulls/1335341197/1

Misiu commented:
@sta any updates?
I've just used VS upgrade assistant and migrated your project to .net standard 2.1. I'll do some more tests, but for now, it looks like everything works fine.
Are you interested in another PR? Or maybe one of the already existing could get merged?
